src/main/java/com/epam/esm/service/CertificateService.java

```diff
@@ -0,0 +1,73 @@
+package com.epam.esm.service;
+
+import com.epam.esm.entities.GiftCertificate;
+import com.epam.esm.repository.Repository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class CertificateService {
+
+    private Repository<GiftCertificate> certificateRepository;
+
+    @Autowired
+    public void setCertificateRepository(Repository<GiftCertificate> certificateRepository) {
+        this.certificateRepository = certificateRepository;
+    }
+
+    public ResponseEntity<String> addCertificate(GiftCertificate giftCertificate) {
```

Comment:
Don`u use ResponseEntity in the service. This is redundant.

Reply:
Ok. Now it returns nothing or throws an Exception